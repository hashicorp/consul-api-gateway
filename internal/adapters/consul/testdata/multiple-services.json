{
  "Name": "multiple-services",
  "Hostnames": [
    "example.com"
  ],
  "Rules": [
    {
      "Matches": [
        {
          "Type": 1,
          "Path": {
            "Type": 1,
            "Value": "/prefix"
          }
        }
      ],
      "Filters": [
        {
          "Type": 0,
          "Header": {
            "Set": {
              "x-set": "1"
            },
            "Add": {
              "x-add": "2"
            },
            "Remove": [
              "x-remove"
            ]
          }
        },
        {
          "Type": 0,
          "Header": {
            "Set": {
              "x-set-too": "1"
            },
            "Add": {
              "x-add-too": "2"
            }
          }
        }
      ],
      "Services": [
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": 0,
              "Header": {
                "Set": {
                  "x-set": "3"
                },
                "Add": {
                  "x-add": "4"
                },
                "Remove": [
                  "x-remove-a"
                ]
              }
            }
          ]
        },
        {
          "Service": {
            "ConsulNamespace": "namespace",
            "Service": "another-service"
          },
          "Weight": 1,
          "Filters": [
            {
              "Type": 0,
              "Header": {
                "Add": {
                  "x-add": "4"
                },
                "Remove": [
                  "x-remove-b"
                ]
              }
            }
          ]
        }
      ]
    }
  ]
}