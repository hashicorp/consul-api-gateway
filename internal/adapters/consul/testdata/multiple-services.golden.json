{
  "Router": {
    "Kind": "service-router",
    "Name": "multiple-services",
    "Namespace": "k8s",
    "Routes": [
      {
        "Match": {
          "HTTP": {
            "PathExact": "/prefix"
          }
        },
        "Destination": {
          "Service": "multiple-services-0",
          "Namespace": "k8s",
          "RequestHeaders": {
            "Add": {
              "x-add": "2",
              "x-add-too": "2"
            },
            "Set": {
              "x-set": "1",
              "x-set-too": "1"
            },
            "Remove": [
              "x-remove"
            ]
          }
        }
      }
    ],
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "Splitters": [
    {
      "Kind": "service-splitter",
      "Name": "multiple-services-0",
      "Namespace": "k8s",
      "Splits": [
        {
          "Weight": 50,
          "Service": "service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Set": {
              "x-set": "3"
            },
            "Remove": [
              "x-remove-a"
            ]
          }
        },
        {
          "Weight": 50,
          "Service": "another-service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Remove": [
              "x-remove-b"
            ]
          }
        }
      ],
      "CreateIndex": 0,
      "ModifyIndex": 0
    }
  ]
}